app/elm/Update.elm

```diff
@@ -1,27 +1,65 @@
-module Update exposing (..)
+port module Update exposing (..)
 
 import Model exposing (..)
-import Data.Request exposing (getResults)
+import Data.Request exposing (getEvents)
+import Data.Location exposing (..)
+import Data.Dates exposing (..)
+import Date exposing (..)
+
+
+port updateMarkers : List Marker -> Cmd msg
 
 
 update : Msg -> Model -> ( Model, Cmd Msg )
 update msg model =
     case msg of
-        UpdatePostcode postcode ->
+        SetPostcode postcode ->
             { model | postcode = postcode } ! []
 
         SetDate date ->
-            { model | date = date } ! []
+            let
+                noDateSelected =
+                    { model | selectedDate = "" }
+
+                newDateSelected =
+                    { model | selectedDate = date }
+            in
+                if model.selectedDate == date then
+                    noDateSelected ! [ updateMarkers (eventMarkers (filterByDate noDateSelected noDateSelected.events)) ]
+                else
+                    newDateSelected ! [ updateMarkers (eventMarkers (filterByDate newDateSelected newDateSelected.events)) ]
 
-        GetSearchResults ->
-            model ! [ getResults ]
+        GetEvents ->
+            model ! [ getEvents ]
 
-        SearchResults (Ok results) ->
-            { model | events = results } ! []
+        Events (Ok events) ->
+            { model | events = events } ! [ updateMarkers (eventMarkers events) ]
 
-        SearchResults (Err err) ->
+        Events (Err err) ->
             let
                 log =
                     Debug.log "Request Error" err
             in
                 model ! []
+
+        GetLocation ->
+            model ! [ getLocation ]
+
+        Location (Ok location) ->
+            { model | userLocation = Just (getCoords location) } ! []
+
+        Location (Err err) ->
+            let
+                log =
+                    Debug.log "Location Error" err
+            in
+                model ! []
+
+        CurrentDate currentDate ->
+            { model | currentDate = Just (fromTime currentDate) } ! []
+
+
+eventMarkers : List Event -> List Marker
```

Comment:
I think moving the ports into their own file would also solve the problem of where to put this function too